### ISSUE TITLE
Missing loaders after update to 12.0.0

### ISSUE THREAD

**Issue by vitorcamachoo** [nrwl/nx#5289]
## Current Behavior

After upgrading nwrl packages using `nx migrate latest`, it started to throw errors on application related with missing loaders, more specifically relates with nullish operators (??).
<img width="586" alt="Screenshot 2021-04-09 at 11 15 49" src="https://user-images.githubusercontent.com/20595956/114165886-f3051100-9924-11eb-9774-6c59580768ff.png">

## Steps to Reproduce

`nx migrate latest`
`npm install --legacy-peer-deps`
`npm run start [application]`

This issue may not be prioritized if details are not provided to help us reproduce the issue.

### Failure Logs

```shell
File was processed with these loaders:
 * ../../node_modules/next/dist/build/webpack/loaders/next-babel-loader.js
You may need an additional loader to handle the result of these loaders.
|   }, [props.onSubscribe, newsLetterRef]);
|   return /*#__PURE__*/_jsxDEV(RoundedPaper, {
>     curvature: props.curvature ?? 'outerTop',
|     className: clsx(classes.root, props.className),
|     children: /*#__PURE__*/_jsxDEV(Grid, {
```

**Comment by mbrainiac** [comment]
Same here 

```
Module build failed (from /Volumes/test/node_modules/url-loader/dist/cjs.js):
Error: Cannot find module 'file-loader'
Require stack:
- /Volumes/test/node_modules/url-loader/dist/index.js
- /Volumes/test/node_modules/url-loader/dist/cjs.js
- /Volumes/test/node_modules/loader-runner/lib/loadLoader.js
- /Volumes/test/node_modules/loader-runner/lib/LoaderRunner.js
- /Volumes/test/node_modules/@nrwl/web/node_modules/webpack/lib/NormalModule.js
- /Volumes/test/node_modules/@nrwl/web/node_modules/webpack/lib/NormalModuleFactory.js
- /Volumes/test/node_modules/@nrwl/web/node_modules/webpack/lib/Compiler.js
- /Volumes/test/node_modules/@nrwl/web/node_modules/webpack/lib/webpack.js
- /Volumes/test/node_modules/@nrwl/web/src/builders/dev-server/dev-server.impl.js
- /Volumes/test/node_modules/@nrwl/tao/src/shared/workspace.js
- /Volumes/test/node_modules/@nrwl/tao/src/commands/run.js
- /Volumes/test/node_modules/@nrwl/tao/index.js
- /Volumes/test/node_modules/@nrwl/cli/lib/run-cli.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:815:15)
    at Function.Module._load (internal/modules/cjs/loader.js:667:27)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.loader (/Volumes/test/node_modules/url-loader/dist/index.js:66:20),
```

**Comment by ptrhck** [comment]
Same here

```
Module parse failed: Unexpected token (136:38)
File was processed with these loaders:
 * ../../node_modules/next/dist/build/webpack/loaders/next-babel-loader.js
You may need an additional loader to handle the result of these loaders.
| 
| export function initializeApollo(initialState = null) {
>   const _apolloClient = apolloClient ?? createApolloClient()
```

**Comment by ptrhck** [comment]
Any ideas yet?

**Comment by nickroberts** [comment]
I was able to fix it for now, by getting rid of optional chaining and nullish coalescing in my apollo client.

```typescript
const _apolloClient = apolloClient ? apolloClient : createApolloClient();
```

```typescript
if (pageProps && pageProps.props) {
  pageProps.props[APOLLO_STATE_PROP_NAME] = client.cache.extract();
}
```
That being said, I shouldn't have to do that. Not sure if it's their custom babel loader config, or what. Odd. `create-next-app` with the apollo template works just fine.

**Comment by aiji42** [comment]
The same error occurs when importing libs using `@nrwl/react/babel` from nextjs apps using `@nrwl/next/babel`.  
I changed the `.balelrc` of the target libs to use `@nrwl/next/babel`, and successfully avoided the problem.  
```
{
  "presets": ["@nrwl/next/babel"],
  "plugins": []
}
```
(By the way, this error is also reproduced in v11.6.x before v12)
---
If you use emotion's css props, you can do this. The `@emotion/babel-plugin` did not work correctly.
```
# yarn install -D @emotion/babel-preset-css-prop
{
  "presets": ["@nrwl/next/babel", "@emotion/babel-preset-css-prop"],
  "plugins": []
}
```

**Comment by ptrhck** [comment]
Upgrading to `12.0.3` resolved the issue for me.

**Comment by aiji42** [comment]
Yeah, I also updated to v12.0.4 and that fixed the problem.

**Comment by charlieargue** [comment]
Hello! Help!

I'm on `12.0.7` and running into this issue (again) now... I thought I had fully fixed it yesterday by adding `import 'regenerator-runtime/runtime';` to each of my next `/pages` but apparently not because it's back today...

I've just done an `nx migrate latest` workflow also hoping that would fix it, but no...

✅ Partial SOLUTION!?: I was able to get things back up and running but I had to add:
```
import 'regenerator-runtime/runtime';
```
* to almost every file in my projects!
* NOTE: this is after switching `@nrwl/react/babel` in my libs to `@nrwl/next/babel` like @aiji42 said above
* ALSO: now I'm also having **Allocation failed - JavaScript heap out of memory** issues during linting my `react-ui` library, sigh...
* as well as seeing RED everywhere (because I assume of same linting issues?) -- all my various library imports seem to have broken?
* TYPESCRIPT version: 4.0.3

Here's my repo, you can easily reproduce it by running the front-end with the **mocked back-end**:
```
# 1) clone the repo
git clone https://github.com/charlieargue/multi-cart.git

# 2) install libraries and fire it up!
cd multi-cart && yarn && yarn mock
```

You can try logging in, but won't be able to do much after that -- it's still a work-in-progress, please disregard the mess!
- username: mocked-user
- password: mocking

IN SUMMARY:
- so I figured out how to get around this **Missing loaders** issue by putting `regenerator-runtime` everywhere
- but switching to `@nrwl/next/babel` in my libs seems to have crashed my linting and all my `@multicart/LIB` imports are erroring out/RED (or it's my 12.0.6/now 7 upgrades?)
- also, I'm getting out-of-memory issues during linting... but that's neither here nor there, sorry...

Here's a loom of what I'm hoping you'll see on the browser side:
- https://www.loom.com/share/c33c4e6c4c024cabbdf144272d1e1721 (~30 seconds)

And here's what I'm seeing in vscode:
- https://www.loom.com/share/e445f08d657c4c5c87c5324126654e92 (~30 seconds)

Thanks so much!

**Comment by parksj10** [comment]
can confirm this is still an issue with 12.0.8 (when upgrading from 11.3.0 using document `nx migrate latest` procedure)...

errors:
```
Error: Can't resolve '@emotion/styled/base' in path/to/story`

`Child HtmlWebpackCompiler:
                          Asset      Size               Chunks  Chunk Names
    __child-HtmlWebpackPlugin_0  6.18 KiB  HtmlWebpackPlugin_0  HtmlWebpackPlugin_0
    Entrypoint HtmlWebpackPlugin_0 = __child-HtmlWebpackPlugin_0
    [./node_modules/html-webpack-plugin/lib/loader.js!./node_modules/@storybook/core/dist/server/templates/index.ejs] 1.97 KiB {HtmlWebpackPlugin_0} [built]

WARN Broken build, fix the error above.
WARN You may need to refresh the browser.
```


Fix:

update `.babelrc` similar to @aiji42 suggestion:

```
{
  "presets": ["@nrwl/react/babel", "@emotion/babel-preset-css-prop"],
  "plugins": []
}
```
